### PR TITLE
Add helm chart

### PR DESCRIPTION
**Pull request by adrianchiris**
- add helm chart
- add automation and release workflow for helm chart

### CODE REVIEW

File: .github/workflows/chart-push-release.yaml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+          GITHUB_TAG: ${{ github.ref_name }}
+          GITHUB_REPO_OWNER: ${{ github.repository_owner }}
+        run: make chart-prepare-release
+    
+      - name: push chart
+        env:
+          GITHUB_TAG: ${{ github.ref_name }}
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          GITHUB_REPO_OWNER: ${{ github.repository_owner }}
+        run: make chart-push-release
```
Comment (ykulazhenkov) [inline .github/workflows/chart-push-release.yaml:28]: nit: no new line at the end

File: Makefile [hunk 2]
```diff
@@ -91,8 +91,134 @@ COVER_MODE = atomic
...
+YQ := $(abspath $(LOCALBIN)/yq)
+YQ_VERSION=v4.44.1
+.PHONY: yq
+yq: $(YQ) ## Download yq locally if necessary.
+$(YQ): | $(LOCALBIN)
+	@curl -fsSL -o $(YQ) https://github.com/mikefarah/yq/releases/download/$(YQ_VERSION)/yq_linux_amd64 && chmod +x $(YQ)
+
+GOLANGCI_LINT = $(LOCALBIN)/golangci-lint
+GOLANGCI_LINT_VERSION ?= v1.59.1
+golangci-lint:
```
Comment (ykulazhenkov) [inline Makefile:217]: missing `.PHONY` entry for golangci-lint

File: Makefile [hunk 3]
```diff
@@ -91,8 +91,134 @@ COVER_MODE = atomic
...
+		HELM_INSTALL_DIR=$(LOCALBIN) USE_SUDO=false $(LOCALBIN)/get_helm.sh && \
+		rm -f $(LOCALBIN)/get_helm.sh; \
+	}
+
+YQ := $(abspath $(LOCALBIN)/yq)
+YQ_VERSION=v4.44.1
+.PHONY: yq
+yq: $(YQ) ## Download yq locally if necessary.
+$(YQ): | $(LOCALBIN)
+	@curl -fsSL -o $(YQ) https://github.com/mikefarah/yq/releases/download/$(YQ_VERSION)/yq_linux_amd64 && chmod +x $(YQ)
```
Comment (ykulazhenkov) [inline Makefile:213]: Here we always download x86 binary. Do we want to download the binary that matches the arch of the build host?

File: deployment/maintenance-operator-chart/templates/metrics-service.yaml [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    app.kubernetes.io/part-of: maintenance-operator
+    control-plane: {{ .Release.Name }}-controller-manager
+  {{- include "maintenance-operator.labels" . | nindent 4 }}
+spec:
+  type: {{ .Values.metricsService.type }}
+  selector:
+    control-plane: {{ .Release.Name }}-controller-manager
+  {{- include "maintenance-operator.selectorLabels" . | nindent 4 }}
+  ports:
+	{{- .Values.metricsService.ports | toYaml | nindent 2 }}
```
Comment (ykulazhenkov) [inline deployment/maintenance-operator-chart/templates/metrics-service.yaml:18]: No new line at the end + mixing of spaces and tabs?

File: deployment/maintenance-operator-chart/templates/operatorconfig.yaml [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+metadata:
+  name: default
+  namespace: {{ .Release.Namespace }}
+  labels:
+    app.kubernetes.io/component: config
+    app.kubernetes.io/created-by: maintenance-operator
+    app.kubernetes.io/part-of: maintenance-operator
+  {{- include "maintenance-operator.labels" . | nindent 4 }}
+spec:
+  logLevel: info
```
Comment (ykulazhenkov) [inline deployment/maintenance-operator-chart/templates/operatorconfig.yaml:12]: nit: no new line at the end

File: deployment/maintenance-operator-chart/templates/webhook.yaml [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+    app.kubernetes.io/created-by: maintenance-operator
+    app.kubernetes.io/part-of: maintenance-operator
+  {{- include "maintenance-operator.labels" . | nindent 4 }}
+spec:
+  type: {{ .Values.webhookService.type }}
+  selector:
+    control-plane: {{ .Release.Name }}-controller-manager
+  {{- include "maintenance-operator.selectorLabels" . | nindent 4 }}
+  ports:
+	{{- .Values.webhookService.ports | toYaml | nindent 2 }}
```
Comment (ykulazhenkov) [inline deployment/maintenance-operator-chart/templates/webhook.yaml:18]: tabs/spaces mixing

File: deployment/maintenance-operator-chart/crds/maintenanceoperatorconfig-crd.yaml
Comment (ykulazhenkov) [inline deployment/maintenance-operator-chart/crds/maintenanceoperatorconfig-crd.yaml:1]: I think we should automatically update CRDs in the helm chart's dir with `make manifests`. WDYT?

File: Makefile (hunk 3 above)
Comment (adrianchiris) [reply Makefile:213]: i dont think its needed. its intended to be run in GH runner.

File: deployment/maintenance-operator-chart/crds/maintenanceoperatorconfig-crd.yaml
Comment (adrianchiris) [reply deployment/maintenance-operator-chart/crds/maintenanceoperatorconfig-crd.yaml:1]: makes sense.

also i will add some additional values for operator config CR in the templates.

**Review by ykulazhenkov — APPROVED** [review]

**Pull request merged**